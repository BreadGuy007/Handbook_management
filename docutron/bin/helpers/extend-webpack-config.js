/**
 * External Dependencies
 */
const path = require( 'path' );

module.exports = function( webpackConfig, usersCwd, docsFolder ) {
	// Adding "docutron" alias
	webpackConfig.resolve.alias.docutron = path.resolve( __dirname, '../../src/config/' );

	// Loading the config folder
	webpackConfig.resolve.alias.config = path.resolve( usersCwd, docsFolder );
	webpackConfig.resolve.modules = webpackConfig.resolve.modules.concat( [ webpackConfig.resolve.alias.config ] );

	// Using the user's node_modules
	const usersNodeModules = path.resolve( usersCwd, 'node_modules' );
	webpackConfig.resolve.modules = webpackConfig.resolve.modules.concat( [ usersNodeModules ] );

	// Deleting CRA scoping
	webpackConfig.resolve.plugins = [];
	webpackConfig.module.rules.forEach( ( rule ) => {
		if ( rule.include ) {
			rule.include = [ rule.include, webpackConfig.resolve.alias.config ];
		}
	} );

	// Adding the markdown loader and exclude if from the file loader
	webpackConfig.module.rules.forEach( rule => {
		if ( rule.loader === require.resolve( 'file-loader' ) ) {
			rule.exclude.push( /\.md/ );
		}
	} );
	webpackConfig.module.rules.push( {
		test: /\.md/,
		use: require.resolve( 'raw-loader' ),
	} );
};
