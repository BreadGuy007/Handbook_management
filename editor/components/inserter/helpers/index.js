/**
 * Helpers associated with the Inserter Component.
 *
 * This file contains a group of functions that help in managing the inserter.
 */

export const isShownBlock = state => block => block.title.toLowerCase().indexOf( state.filterValue.toLowerCase() ) !== -1;

export const listVisibleBlocksByCategory = ( state ) => ( blocks ) => {
	return blocks.reduce( ( groups, block ) => {
		if ( ! isShownBlock( state )( block ) ) {
			return groups;
		}
		if ( ! groups[ block.category ] ) {
			groups[ block.category ] = [];
		}
		groups[ block.category ].push( block );
		return groups;
	}, {} );
};

const flattenObjectIntoArray = ( object ) => {
	let list = [];
	for ( const key in object ) {
		list.push( object[ key ] );
	}
	list = list.reduce( ( values, objectValue ) => values.concat( objectValue ), [] );

	return list;
};

const nextFocusableRef = ( component ) => {
	const visibleBlocksByCategory = flattenObjectIntoArray( listVisibleBlocksByCategory( component.state )( component.blockTypes ) );
	const focusables = visibleBlocksByCategory.map( blockType => blockType.slug ).concat( 'search' );

	if ( null === component.state.focusedElementRef ) {
		return focusables[ 0 ];
	}

	const currentIndex = focusables.findIndex( ( elementRef ) => elementRef === component.state.focusedElementRef );
	const nextIndex = currentIndex + 1;
	const highestIndex = focusables.length - 1;

	// Check boundary so that the index is does not exceed the length.
	if ( nextIndex > highestIndex ) {
		// Cycle back to other end.
		return focusables[ 0 ];
	}

	return focusables[ nextIndex ];
};

const previousFocusableRef = ( component ) => {
	const visibleBlocksByCategory = flattenObjectIntoArray( listVisibleBlocksByCategory( component.state )( component.blockTypes ) );
	const focusables = visibleBlocksByCategory.map( blockType => blockType.slug ).concat( 'search' );

	// Initiate the menu with the first block.
	if ( null === component.state.focusedElementRef ) {
		return focusables[ 0 ];
	}

	const currentIndex = focusables.findIndex( ( elementRef ) => elementRef === component.state.focusedElementRef );
	const nextIndex = currentIndex - 1;
	const lowestIndex = 0;

	// Check boundary so that the index is does not exceed the length.
	if ( nextIndex < lowestIndex ) {
		// Cycle back to other end.
		return focusables[ focusables.length - 1 ];
	}

	return focusables[ nextIndex ];
};

export const focusNextFocusableRef = ( component ) => {
	const next = nextFocusableRef( component );
	component.changeMenuSelection( next );
};

export const focusPreviousFocusableRef = ( component ) => {
	const previous = previousFocusableRef( component );
	component.changeMenuSelection( previous );
};
