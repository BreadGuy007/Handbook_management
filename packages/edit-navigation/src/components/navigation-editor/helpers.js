export const flattenBlocks = ( blocks ) =>
	blocks.flatMap( ( item ) =>
		[ item ].concat( flattenBlocks( item.innerBlocks || [] ) )
	);
