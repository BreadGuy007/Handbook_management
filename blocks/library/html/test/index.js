/**
 * Internal dependencies
 */
import '../';
import { blockEditRender } from 'blocks/test/helpers';

describe( 'core/html', () => {
	test( 'block edit matches snapshot', () => {
		const wrapper = blockEditRender( 'core/html' );

		expect( wrapper ).toMatchSnapshot();
	} );
} );
