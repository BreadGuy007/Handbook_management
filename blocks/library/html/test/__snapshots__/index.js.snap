// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core/html block edit matches snapshot 1`] = `
<textarea
  aria-label="HTML"
  class="blocks-plain-text wp-block-html"
  rows="1"
/>
`;
